sdk/storage/Azure.Storage.Common/src/GeoRedundantReadPolicy.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+using System;
+using System.Net;
+using Azure.Core.Pipeline;
+
+namespace Azure.Storage.Common
+{
+    internal class GeoRedundantReadPolicy : SynchronousHttpPipelinePolicy
+    {
+        private const string NextHostKey = "Azure.Storage.Common.NextHost";
+        private const string ResourceNotReplicatedToSecondaryStorageKey = "Azure.Storage.Common.ResourceNotReplicatedToSecondaryStorage";
+
+        private readonly string _secondaryStorageHost;
+
+        public GeoRedundantReadPolicy(Uri secondaryStorageUri) => this._secondaryStorageHost = secondaryStorageUri.Host;
+
+        public override void OnSendingRequest(HttpPipelineMessage message)
+        {
+            if (this._secondaryStorageHost == null)
+            {
+                return;
+            }
+            if (message.Request.Method != RequestMethod.Get && message.Request.Method != RequestMethod.Head)
+            {
+                return;
+            }
+            var nextHost = message.TryGetProperty(NextHostKey, out var previousHostObj) ? previousHostObj as string : null;
+            if (nextHost == null) // first time being called for the message, so we won't toggle the host
+            {
+                message.SetProperty(NextHostKey, this._secondaryStorageHost); // queue up the secondary host for subsequent retries
+                return;
+            }
+            var resourceNotReplicatedToSecondaryStorage = !message.TryGetProperty(ResourceNotReplicatedToSecondaryStorageKey,
```

Comment:
Could simplify to `message.TryGetProperty(ResourceNotReplicatedToSecondaryStorageKey, out var value) && (bool)value;`